prover/src/lib.rs

```diff
@@ -568,14 +581,19 @@ pub trait Prover {
         &self,
         trace: &Self::Trace,
         domain: &StarkDomain<Self::BaseField>,
+        num_partitions: usize,
         channel: &mut ProverChannel<'_, Self::Air, E, Self::HashFn, Self::RandomCoin, Self::VC>,
     ) -> (Self::TraceLde<E>, TracePolyTable<E>)
```

Comment:
Why do we need to pass `num_partitions` into this function? Shouldn't we already have access to it via `self.options()`?